### ISSUE TITLE
empty strings in struct and list are converted to NULLs

### ISSUE THREAD

**Issue by wzzzzd** [risingwavelabs/risingwave#4521]
**Describe the bug**
```
dev=> create table t1 (a varchar[]);
CREATE_TABLE
dev=> insert into t1 values (Array['abc',NULL,'']);
INSERT 0 1
dev=> select * from t1;
        a        
-----------------
 {abc,NULL,NULL}
(1 row)

dev=> create table t2 (a struct<v1 int, v2 varchar>);
CREATE_TABLE
dev=> insert into t2 values ((1, '')), ((2, NULL));
INSERT 0 2
dev=> select * from t2;
    a     
----------
 (2,NULL)
 (1,NULL)
(2 rows)
```

**To Reproduce**
As above.

**Expected behavior**
Empty strings should not be converted to NULLs.

**Comment by jlerche** [comment]
I've been going through this bug, but it's too hard to tell if it's due to the serialization or deserialization, at the very least the null value seems to occur here https://github.com/singularity-data/risingwave/blob/e9a0a7b7cc6d07ba55361a4fa63203fab79ac21a/src/common/src/array/list_array.rs#L345

**Comment by lmatz** [comment]
> I've been going through this bug, but it's too hard to tell if it's due to the serialization or deserialization, at the very least the null value seems to occur here https://github.com/singularity-data/risingwave/blob/e9a0a7b7cc6d07ba55361a4fa63203fab79ac21a/src/common/src/array/list_array.rs#L345 

Yes, I believe by the same reasoning in https://github.com/singularity-data/risingwave/issues/4672, we should only use either memcomparable encoding or value encoding in the future.

**Comment by jlerche** [comment]
> Yes, I believe by the same reasoning in #4672, we should only use either memcomparable encoding or value encoding in the future.

Makes sense, might as well close this issue then in favor of #4672

**Comment by xiangjinwu** [comment]
We still need to confirm if the queries above are actually fixed after #4672, so keeping this open

**Comment by fuyufjh** [comment]
Any updates?

**Comment by lmatz** [comment]
> Any updates?

Require #4672 at least, and #4672 is put into the next release.

**Comment by waruto210** [comment]
solved by #5880
